src/api/app/models/project.rb

```diff
@@ -1257,12 +1257,14 @@ def bsrequest_repos_map(project)
 
   private :bsrequest_repos_map
 
+  # NOTE: This has to cover project name validations in backend/BSVerify.pm (verify_projid)
   def self.valid_name?(name)
     return false unless name.is_a?(String)
     return false if name == '0'
-    return false if name =~ /::/
+    return false if name =~ /:[:\._]/
+    return false if name =~ /\A[:\._]/
     return false if name.end_with?(':')
-    return true if name =~ /\A[a-zA-Z0-9][-+\w\.:]{0,199}\z/
+    return true  if name =~ /\A[\w-+\.:]+\z/
```

Comment:
.valid_name? is used directly all over the place, outside of validations.

Reply:
And that is an issue because... ?

Reply:
It will now return true for projects with a name longer than 200 characters? :-)

Reply:
Good point. Next time please just say that;-)

Reply:
backend would need to be changed as well, or this would just crash.
But has this been discussed with backend people?

Reply:
@adrianschroeter The validation is already in the backend

Reply:
@bgeuken yes, with the 200 char limit ... so longer strings will get refused from backend, even when you can create them in the frontend.

Reply:
And also in the [frontend](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/project.rb#L123)...